packages/aws-library/src/aws_library/s3/_client.py

```diff
@@ -424,7 +425,16 @@ async def copy_object(
                     bytes_transfered_cb, file_name=f"{dst_object_key}"
                 )
             }
+        # NOTE: boto3 copy function uses copy_object until 'multipart_threshold' is reached then switches to multipart copy
+        # copy_object does not provide any callbacks so we can't track progress so we need to ensure at least the completion
+        # of the object is tracked
         await self._client.copy(**copy_options)
+        if bytes_transfered_cb:
+            if object_metadata is None:
+                object_metadata = await self.get_object_metadata(
+                    bucket=bucket, object_key=dst_object_key
+                )
+            bytes_transfered_cb(object_metadata.size, file_name=f"{dst_object_key}")
```

Comment:
What about also reporting when it starts?

Reply:
there is some kind of filtering when it updates in the end. so it is not necessary. I do not want to overcomplicate it.